.github/workflows/ci.yml

```diff
@@ -98,3 +96,23 @@ jobs:
         with:
           command: fmt
           args: --all -- --check
+
+  clippy:
+    if: github.event.pull_request.draft == false
+
+    name: Clippy lint checks
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+          components: clippy
```

Comment:
I think this should be left as `override: false`. The point of keeping the library at `nightly` was to allow us to use unstable features if needed. If the CI is not using `nightly` then we cannot use it as a valid test, since it isn't using the same tools we are using.

If we use the `rust-toolchain.toml` file I suggested, then `actions-rs` should automatically download the right tools, and also it enforces that when developing, maintainers of this project have at the very least `rustfmt` and `clippy` installed, which are requirements of the CI for merging anyway.

Reply:
If we don't want to change the `rust-toolchain` then we can keep this CI as
```yaml
with:
  override: false
  components: clippy
```

Reply:
Makes more sense. I agree.

Wasn't aware about these latest changes in the rust-toolchain file stuff!

Thanks for the update!

Reply:
Sorry, it looks like they didn't finish upgrading to the new version: https://github.com/actions-rs/toolchain/issues/126. I usually don't use `actions-rs`.

Reply:
Exactly.  I was just going to say that. 
Seeing https://github.com/actions-rs/toolchain#the-toolchain-file seems that the naming convention differs. But supports the same date format.

Reply:
Yeah, so I would keep the `rust-toolchain` the same and just have `clippy` in the YAML as above.

Reply:
Yepp! Agree! Resetting!